src/scripts/create_fullstack_ner.py

```diff
@@ -0,0 +1,180 @@
+
+def parse_conllu_data(raw_data: str) -> List[Dict[str, Union[List[str], str]]]:
+    """Parse CoNLL-U format data and return a list of sentence records."""
+    records = []
+    lines = raw_data.strip().split("\n")
+
+    # Initialize data dictionary for current sentence
+    data_dict: Dict[str, List[str]] = defaultdict(list)  # type: ignore[assignment]
+
+    for line in lines:
+        line = line.strip()
+
+        # Skip comments (lines starting with #)
+        if line.startswith("#"):
+            continue
+
+        # Empty line indicates end of sentence
+        elif line == "":
+            if len(data_dict["tokens"]) > 0:
+                # Create record with tokens, labels, and text
+                record = {
+                    "tokens": data_dict["tokens"],
+                    "labels": data_dict["labels"],
+                    "text": " ".join(data_dict["tokens"]),
+                }
+                records.append(record)
+            # Reset for next sentence
+            data_dict = defaultdict(list)
+
+        # Parse token line
+        else:
+            columns = line.split("\t")
+            if len(columns) >= 7:  # Ensure we have enough columns
+                token_id = columns[0]
+                token = columns[1]
+                # Use column 6 for NER tags (0-indexed, so columns[6] is column 7)
+                ner_tag = columns[6] if len(columns) > 6 else "O"
+
+                # Skip multi-word tokens (those with "-" in ID)
+                if "-" not in token_id and "." not in token_id:
+                    data_dict["tokens"].append(token)
+                    data_dict["labels"].append(ner_tag)
+
+    # Handle last sentence if data doesn't end with empty line
+    if len(data_dict["tokens"]) > 0:
+        record = {
+            "tokens": data_dict["tokens"],
+            "labels": data_dict["labels"],
+            "text": " ".join(data_dict["tokens"]),
+        }
+        records.append(record)
+
+    return records  # type: ignore[return-value]
+
+
+def load_fullstack_data() -> List[Dict[str, Union[List[str], str]]]:
+    """Load and parse all FullStack NER data from the local directory."""
+    # Path to the local data directory
+    data_dir = "../FullStack/NamedEntities/data"
+    if not os.path.exists(data_dir):
+        raise FileNotFoundError(
+            f"Required data directory '{data_dir}' not found. "
+            "Please clone the FullStack repository and ensure the data "
+            "is available at this path."
+        )
+
+    # Find all .conll2003 files in the directory
+    conll_files = glob.glob(os.path.join(data_dir, "*.conll2003"))
+
+    all_records = []
+
+    for file_path in sorted(conll_files):
+        # Read the file
+        with open(file_path, "r", encoding="utf-8") as f:
+            content = f.read()
+
+        # Parse the CoNLL-U data
+        records = parse_conllu_data(content)
+        all_records.extend(records)
+
+    return all_records
+
+
+def main() -> None:
```

Comment:
[nit] Main function should ideally be at the top of the script, and then the other functions in the other they're used.

Reply:
Same thing with the other scripts.